service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,70 @@
         </actions>
     </service>
 
+    <service verb="get" noun="AutoPackageInfoList">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="shipmentSourceList">
+                    <econdition field-name="orderId" from="itemInfoMap.orderId"/>
+                </entity-find>
+                <iterate list="shipmentSourceList" entry="shipmentSourceMap">
+                    <set field="shipmentId" from="shipmentSourceMap.shipmentId"/>
```

Comment:
What if an order has multiple shipments. Can you improve code to handle multiple shipment Ids.
Also please check if we can use  ShipmentAndItemSource view entity?

Reply:
But then it has different OrderPartSeqId in orderPart entity, how to deal with that?